test/unit/lualib/lualib.spec.ts

```diff
@@ -377,6 +377,39 @@ test.each([
     expect(result).toBe(JSON.stringify(array.sort(compareFn)));
 });
 
+test.each([
+    { array: [1, [2, 3], 4], expected: [1, 2, 3, 4] },
+    { array: [1, [2, 3], 4], depth: 0, expected: [1, [2, 3], 4] },
+    { array: [1, [[2], [3]], 4], expected: [1, [2], [3], 4] },
+    { array: [1, [[[2], [3]]], 4], depth: Infinity, expected: [1, 2, 3, 4] },
+])("array.flat (%p)", ({ array, depth, expected }) => {
+    // TODO: Remove once `Infinity` would be implemented
+    const luaDepth = depth === Infinity ? "1 / 0" : depth;
+    const result = util.transpileAndExecute(`
+        return JSONStringify(${JSON.stringify(array)}.flat(${luaDepth}))
+    `);
+
+    expect(JSON.parse(result)).toEqual(expected);
```

Comment:
Might be nicer to compare this to `array.flat()` instead of `expected`

Reply:
https://node.green/#ES2019-features-Array-prototype--flat--flatMap-